app/Exceptions/Store/FulfillmentException.php

```diff
@@ -3,10 +3,10 @@
 // Copyright (c) ppy Pty Ltd <contact@ppy.sh>. Licensed under the GNU Affero General Public License v3.0.
 // See the LICENCE file in the repository root for full licence text.
 
-namespace App\Libraries\Fulfillments;
+declare(strict_types=1);
 
-use App\Exceptions\ValidationException;
+namespace App\Exceptions\Store;
 
-class FulfillmentException extends ValidationException
+class FulfillmentException extends OrderException
```

Comment:
was this caught in the controller before? 🤔 looks like the listeners are dispatched in sync and thus previously potentially caught by the controller?

Reply:
That's a good point; originally it was planned to have the fulfillers eventually moved to workers so it wasn't intended to have their exceptions caught by the controller, but we never went through with that 🤔

Reply:
I guess it should be handled correctly then? Either queue the event handlers or make such errors return the expected code (200?) so the payment processor receives the correct status.